### PR TITLE
Script to reorder and visualize connectivity matrix

### PR DESCRIPTION
**Pull request by frheault**
This script is in continuity with the 4-5 connectivity scripts I made.

After decomposing a tractogram into its atlas-based combinatorial bundles with `scil_decompose_connectivity.py` #84 
Matrices can be constructed (and weighed) with `scil_compute_connectivity.py` #84 
These matrices can be manipulated with `scil_connectivity_math.py` #86 

This PR enable the user to reorder a connectivity matrix and/or create only a sub-network matrix with `scil_reorder_connectivity.py`
And to visualize the matrix and properly name axis and basic colorbar options with `scil_visualize_connectivity.py`

Here is some testing data:
https://drive.google.com/open?id=1L92ZrRC54U1EhI0IyzSLv5OuotutCiir

### CODE REVIEW

**Comment by frheault** [comment]
@mdesco if you want to quickly try these two branch
_frheault:convert_connectivity_to_sft_ and _frheault:reorder_visu_connectivity_.

They are related to Sami's project and are working together. There is a link to testing data in both.
This would facilitate testing, since they are made to work together your input of using them both one after the other is likely to help a lot with the 'instruction' on how to use them

File: scripts/scil_visualize_connectivity.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                   fontsize=args.axis_text_size[1])
+
+    if args.show_only:
+        plt.show()
+    else:
+        plt.savefig(args.out_png, dpi=300, bbox_inches='tight')
+
+
+if __name__ == "__main__":
+    main()
```
Comment (mdesco) [inline scripts/scil_visualize_connectivity.py]: requirements.txt need to include matplotlib please

./bash_cmd.sh
Traceback (most recent call last):
  File "/Users/desm2239/Research/Source//scilpy/scripts/scil_visualize_connectivity.py", line 22, in <module>
    import matplotlib.pyplot as plt
ModuleNotFoundError: No module named 'matplotlib'
Traceback (most recent call last):
  File "/Users/desm2239/Research/Source//scilpy/scripts/scil_visualize_connectivity.py", line 22, in <module>
    import matplotlib.pyplot as plt
ModuleNotFoundError: No module named 'matplotlib'

Comment (frheault) [reply scripts/scil_visualize_connectivity.py]: I'm going to add it.

Also for the text placement, does reducing the font works ?
I mean, is the quality high enough to be able to read something much smaller ?

Comment (mdesco) [reply scripts/scil_visualize_connectivity.py]: This works well on my end. Maybe you could make this the default or have a heuristic depending on the size of the matrix... thanks!

File: scripts/scil_reorder_connectivity.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+# -*- coding: utf-8 -*-
+
+"""
+Re-order a connectivity matrix using a json file in a format such as:
+    {"temporal": [[1,3,5,7], [0,2,4,6]]}.
+The key is to identify the sub-network, the first list is for the
+column (x) and the second is for the row (y).
+
+The values refers to the coordinates in the matrix, but if the --labels_list
+parameter is used, the values will refers to the label which will be converted
```
Comment (EmmaRenauld) [inline scripts/scil_reorder_connectivity.py]: For people who use the scripts without knowing python. Should we remind them that indices start at 0, not 1?

File: scripts/scil_reorder_connectivity.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+from scilpy.io.utils import (add_overwrite_arg, assert_inputs_exist,
+                             assert_outputs_exist)
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=__doc__)
+
+    p.add_argument('in_matrix',
+                   help='Connectivity matrix in numpy format.')
```
Comment (EmmaRenauld) [inline scripts/scil_reorder_connectivity.py]: explicitly say (.npy) maybe?

File: scripts/scil_visualize_connectivity.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Script to display a connectivity matrix and adjust the desired visualization.
+Made to work with scil_decompose_connectivity.py and
+scil_visualize_connectivity.py.
+
+This scripts can either display the axis labels as:
```
Comment (EmmaRenauld) [inline scripts/scil_visualize_connectivity.py]: This script

File: scripts/scil_visualize_connectivity.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+- Coordinates (0..N)
+- Labels (using --labels_list)
+- Names (using --labels_list and --lookup_table)
+
+If the matrix was made from a bigger matrix using scil_reorder_connectivity.py
```
Comment (EmmaRenauld) [inline scripts/scil_visualize_connectivity.py]: add a comma at the end of line 14

File: scripts/scil_visualize_connectivity.py [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+provide the json and specified the key (using --reorder_json)
```
Comment (EmmaRenauld) [inline scripts/scil_visualize_connectivity.py]: specify

File: scripts/scil_visualize_connectivity.py [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+                             assert_outputs_exist)
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description='Flip one or more axes of the '
+                                            'encoding scheme matrix.')
+
+    p.add_argument('in_matrix',
+                   help='Connectivity matrix in numpy format.')
```
Comment (EmmaRenauld) [inline scripts/scil_visualize_connectivity.py]: (.npy)

File: scripts/scil_reorder_connectivity.py [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+"""
+Re-order a connectivity matrix using a json file in a format such as:
+    {"temporal": [[1,3,5,7], [0,2,4,6]]}.
+The key is to identify the sub-network, the first list is for the
+column (x) and the second is for the row (y).
+
+The values refers to the coordinates in the matrix, but if the --labels_list
+parameter is used, the values will refers to the label which will be converted
+to the appropriate coordinates. This file must be the same as the one provided
+to the scil_decompose_connectivity.py
```
Comment (EmmaRenauld) [inline scripts/scil_reorder_connectivity.py]: Could you give a small example, like you did higher with coordinates?

**Comment by AntoineTheb** [comment]
`scil_reorder_connectivity.py` crashes if a label `json` is used without a label list. Maybe a warning or a proper error message in this case could be added ?

File: scripts/scil_visualize_connectivity.py [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+
+    add_overwrite_arg(p)
+
+    return p
+
+
+def write_values(ax, matrix, properties):
+    width, height = matrix.shape
+    font0 = FontProperties().copy()
+    font0.set_size(properties[0])
```
Comment (AntoineTheb) [inline scripts/scil_visualize_connectivity.py]: Est-ce que ce serait possible de auto-gérer la taille de la police si y'a trop de labels ? Voir l'image de Max

File: scripts/scil_reorder_connectivity.py (hunk 8 above)
Comment (frheault) [reply scripts/scil_reorder_connectivity.py]: Once everything is merged, I will create an example/tutorial separately

**Pull request merged**